Intern project realization

kairoslav opened the pull request.

kairoslav committed:
Added: project

kairoslav committed:
Added: newline at the end of file

kairoslav committed:
Added: newline at the end of file

Aleksandr-Shvets commented on build.gradle:
```diff
@@ -0,0 +1,36 @@
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa:2.5.5'
```
все версии необходимо вынести в переменные

Aleksandr-Shvets replied:
также непонятно почему id 'org.springframework.boot' version '2.5.6'
зависимости 2.5.5

Aleksandr-Shvets commented on src/main/java/com/task/raif/controller/ErrorController.java:
```diff
@@ -0,0 +1,55 @@
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    public ResponseEntity<?> MethodArgumentTypeMismatchExceptionHandler(MethodArgumentTypeMismatchException ex) {
+        String body = "Параметры запроса отсутствуют или имеют некорректный формат: " + ex.getName();
+        return new ResponseEntity<>(Map.of("error", body), HttpStatus.BAD_REQUEST);
```
здесь и ниже - дублирование кода return new ResponseEntity<>(Map.of("error", body), HttpStatus.BAD_REQUEST);

Aleksandr-Shvets commented on src/main/java/com/task/raif/controller/SocksController.java:
```diff
@@ -0,0 +1,47 @@
+
+    @GetMapping("/api/socks")
+    public ResponseEntity<?> getSocksController(
+            @RequestParam @NotBlank String color,
```
непонятно почему используются разные механизмы валидации, старайтесь придерживаться одного стиля в проекте/наборе проектов

Aleksandr-Shvets commented on src/main/java/com/task/raif/controller/SocksController.java:
```diff
@@ -0,0 +1,47 @@
+    @PostMapping("/api/socks/income")
+    public ResponseEntity<?> socksIncome(@Valid @RequestBody Socks socks) {
+        socksService.income(socks);
+        return new ResponseEntity<>(HttpStatus.OK);
```
тк тело ответа пустое, можно сделать метод void и пометить аннотацией @ ResponseStatus

Aleksandr-Shvets commented on src/main/java/com/task/raif/model/Socks.java:
```diff
@@ -0,0 +1,28 @@
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Socks {
```
Логичнее назвать это. SocksDto
а SocksModel. переименовать в Socks тк пакет говорит сам за себя

Aleksandr-Shvets commented on src/main/java/com/task/raif/model/SocksModel.java:
```diff
@@ -0,0 +1,35 @@
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
```
это тут лишнее

Aleksandr-Shvets commented on src/main/java/com/task/raif/service/SocksServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+import com.task.raif.repository.SocksRepository;
+import org.springframework.stereotype.Component;
+
+@Component
```
не ошибка, но вопрос на подумать, а почему не  @Service ?

Aleksandr-Shvets commented on src/test/java/com/task/raif/RaifApplicationTests.java:
```diff
@@ -0,0 +1,13 @@
+class RaifApplicationTests {
+
+    @Test
+    void contextLoads() {
```
хорошее приложение всегда содержит тесты, покрывающие функционал, 
в данном проекте можно было написать тесты на эндпоинты  с использование mockMvc и простые юнит тесты для сервиса

kairoslav committed:
Code refactoring and some improvements.
Added: Swagger documentation, Mock and Unit tests.
Fixed: gradle spring version, code duplication.
Renamed: Socks to SocksDto and SocksModel to Socks.
